src/client/actions/editActions.js

```diff
@@ -158,13 +158,36 @@ export const onChangeArticle = (key, value) => {
   }
 }
 
-export const changeArticle = (key, value) => {
+export const changeArticle = (data) => {
   return {
     type: actions.CHANGE_ARTICLE,
     payload: {
-      key,
-      value
+      data
+    }
+  }
+}
+
+export const changeArticleData = (key, value) => {
+  return (dispatch, getState) => {
+    const { edit: { article } } = getState()
+    let data = {}
+
+    if (typeof key === 'object') {
+    // extend article with an object of keys
+      data = key
+    } else if (key.split('.').length > 1) {
+    // change a nested object value
+      const parentKey = key.split('.')[0]
+      const childKey = key.split('.')[1]
+      const newObject = clone(article[parentKey]) || {}
+
+      newObject[childKey] = value
+      data[parentKey] = newObject
+    } else {
+    // change a single key's value
+      data[key] = value
```

Comment:
Very into this! Following that Backbone set pattern to make it an optional shorthand is nice.